### ISSUE TITLE
API documentation update

### ISSUE THREAD

**Issue by tensor5** [Palmik/data-store#4]
Hi,
data-store seems a very interesting library for databases, but could you improve the documentation?
Explain, for example, what is the role of `tag` in

```
data Store tag krs irs ts v
```

it seems that there is a mistake with the later example

```
...
type ContentStore = Store ContentStoreKRS ContentStoreIRS ContentStoreTS Content
```

Also, it is not clear how the `O` and `M` specifications affect the operations that I can run on a `Store`.

Thanks

**Comment by Palmik** [comment]
I have added explanation of the `tag` type argument of `Store`. I have added the missing `ContentStoreTag` to the example that is part of the documentation of `Data.Store.Internal.Type.Store` (the same example is at the top of the `Data.Store` module description and in the `examples/` directory, but those two instances already contained the `ContentStoreTag`). 

The only functions that depends on specific relation of some dimension are `lookupOrderByA` and `lookupOrderByD` (which is documented). The functions type-check only when it makes sense (see the documentation for more details).
